### PR TITLE
Double edge swaps

### PR DESCRIPTION
**Pull request by nwlandry**
Added the ability to swap the membership of two nodes.

### CODE REVIEW

File: xgi/classes/hypergraph.py [hunk 1]
```diff
@@ -713,6 +713,70 @@ def add_weighted_edges_from(self, ebunch, weight="weight", **attr):
...
+        ----------
+        n_id1 : hashable
+            The ID of the first node, originally a member of the first edge.
+        n_id2 : hashable
+            The ID of the second node, originally a member of the second edge.
+        e_id1 : hashable
+            The ID of the first edge.
+        e_id2 : hashable
+            The ID of the second edge.
+        is_loopy : hashable, default True
```
Comment (leotrs) [inline xgi/classes/hypergraph.py:729]:
```suggestion
        is_loopy : bool, default True
```

File: xgi/classes/hypergraph.py [hunk 2]
```diff
@@ -713,6 +713,70 @@ def add_weighted_edges_from(self, ebunch, weight="weight", **attr):
...
+            Whether edges can be loopy.
+
+        Raises
+        -----
```
Comment (leotrs) [inline xgi/classes/hypergraph.py:733]:
```suggestion
        Raises
        ------
```

**Review by leotrs — APPROVED** [review]

File: xgi/classes/hypergraph.py (hunk 1 above)
Comment (nwlandry) [reply xgi/classes/hypergraph.py]: Fixed.

File: xgi/classes/hypergraph.py (hunk 2 above)
Comment (nwlandry) [reply xgi/classes/hypergraph.py]: Fixed.

**Pull request merged**